### ISSUE TITLE
[Bug]: v9 Avatar: [role='img'] elements have an alternative text (on Avatar root)

### ISSUE THREAD

**Issue by jurokapsiar** [microsoft/fluentui#22659]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
vr-tests storybook
```

### Are you reporting Accessibility issue?

yes

### Reproduction

vr-tests storybook

## Actual Behavior
axe validation fails with an error [role='img'] elements have an alternative text (on Avatar root)

## Expected Behavior
Avatar root needs to have a label or alt

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by smhigley** [comment]
@jurokapsiar for Avatar, we have a kind of unorthodox approach to the image, where the actual `<img>` tag is not exposed and has an empty `alt`, and the parent is a `<span role="img" aria-label="actual name string">`. The reason for this was reconciling several different Avatar variations and components, including the initials, and no-image variants.

The version on our main storybook does not have axe errors, though @behowell would know more about the Avatar vr-tests.

**Comment by behowell** [comment]
@jurokapsiar The issue description mentions that this repros on the vr-tests storybook. However, those weren't written with the intention to pass a11y tests and might be missing aria attributes in some cases; they're just for visual testing and aren't public facing. Are you seeing this on the public docsite?

**Comment by smhigley** [comment]
I think there may also have been some slot wrangling and Persona-related concerns around moving the semantic `img` node to the parent. I've forgotten the details, but we did go through a bunch of options before landing on the current setup 😅

**Comment by smhigley** [comment]
Ignore my comment and just go with @behowell's. I didn't read the issue closely enough, and got `<img alt>` in my head instead of `role="img" element` 😅
